sympy/functions/elementary/complexes.py

```diff
@@ -758,9 +758,17 @@ def eval(cls, arg):
                 break
         else:
             return S.NaN
-        from sympy.functions.elementary.exponential import exp_polar
+        from sympy.functions.elementary.exponential import exp, exp_polar
         if isinstance(arg, exp_polar):
             return periodic_argument(arg, oo)
+        elif isinstance(arg, exp):
+            i_ = im(arg.args[0])
+            if i_.is_comparable:
+                i_ %= 2*S.Pi
+                if i_ > S.Pi:
+                    i_ -= 2*S.Pi
+                return i_
```

Comment:
Maybe this could use the `_pi_coeff` function:
https://github.com/sympy/sympy/blob/4adf8d2417fbb5af0841aa6348bf8344901e8152/sympy/functions/elementary/trigonometric.py#L168-L183

Reply:
I looked at that and `_peel_coeff` and neither looks quite up to the task since this does not require the number to be a multiple of `pi`. I also not that the parity of a symbolic coefficient to `pi` is handled by `mod` with `(e*pi)%(2*pi)` and `e`, even, returns 0 while replacing `e` with `e+1` returns `pi`.

I'll leave for @gschintgen to confirm desire for change or not since I could be missing something.

Reply:
Thanks for reviewing, @smichr and @oscarbenjamin !
I'm not sure `_pi_coeff` is helpful. It's rather the multiple of `I` that's extracted. There's no need for an explicit `pi`. The proposed code also handles e.g.
```
In [1]: arg(exp(4*I))
Out[1]: 4 - 2⋅π
```
I think it should be fine as is.